### ISSUE TITLE
ENH: add tseries.offsets prefix_mapping to public API

### ISSUE THREAD

**Issue by machow** [pandas-dev/pandas#35482]
#### Is your feature request related to a problem?

Currently, a user can use the abbreviation for an offset (e.g. "W" for week). They will see offsets used in the docs, but the mapping from abbreviation to offset (pandas._libs.tslibs.offsets.prefix_mappings) is not public. It doesn't seem possible to answer through the public API: "what offset does this abbreviation correspond to?"

I noticed that there is a [table in the user guide](https://github.com/pandas-dev/pandas/blob/master/doc/source/user_guide/timeseries.rst#dateoffset-objects) that seems to summarize the info contained in prefix_mapping, which is really helpful! However, it would be useful to access it programmatically (e.g. for programmatically listing options to a user).

#### Describe the solution you'd like

What do you think about moving `prefix_mapping` back into pandas.tseries.offsets.prefix_mapping, and making it part of the public API? Based on the docs, it sounds like including it in the "Date offsets" section of the API docs would be sufficient for it to be considered public.

#### API breaking implications

It seems like this action would carry the same risk of API breaking as using the abbreviations in methods like `to_period()` and `as_freq()`, so shouldn't create additional risk (but could be totally wrong here :).

#### Describe alternatives you've considered

The other option would be to put the abbreviation info onto the offset classes themselves, so that the mappings emerge from the classes, but that seems like it'd be a much bigger change!

**Comment by TomAugspurger** [comment]
Makes sense to me. cc @jbrockmendel if you have any thoughts / objections.

**Comment by jbrockmendel** [comment]
No real objections
